### PR TITLE
[DOCKER-405] Switch to images with an embedded tomcat

### PR DESCRIPTION
**Pull request by thijslemmens**
This PR bundles a few changes:
- Drop legacy images
- Drop Alfresco <6.1
- Move to GA for all builds (including enterprise)
- Based on eclipse-temurin
- Using tomcat-embedded
- No longer building on docker-openjdk and docker-tomcat
- Add support for JSON logging
- Add support for access logging (in JSON)
- All logging should go to stdout, no more file logging

I did my best to check projects for what they actually use from docker-java, but when reviewing, please validate it with a project.

### CODE REVIEW

**Comment by thijslemmens** [comment]
I still need to update the changelog, and I see that the JSON logging configuration is not in the README

**Review by RVanhuysseXenit — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
+        java-version: 11
     - name: Grant execute permission for gradlew
       run: chmod +x gradlew
     - name: Write build date
       run: "mkdir -p ~/.gradle && echo 'buildDate='$(date +%s) >> ~/.gradle/gradle.properties"
+    - name: Login to Docker
+      env:
+        DOCKER_USER: ${{ secrets.XENIT_DOCKER_REGISTRY_USERNAME }}
+        DOCKER_PASSWORD: ${{ secrets.XENIT_DOCKER_REGISTRY_PASSWORD }}
+      run: echo "$DOCKER_PASSWORD" | docker login hub.xenit.eu --username "$DOCKER_USER" --password-stdin
```
[the remark this answers is not in the document]
Comment (thijslemmens) [reply .github/workflows/ci.yml:30]: This is the recommended approach: https://github.com/xenit-eu/alfresco-docker-gradle-plugin/blob/master/docs/01-setup.md#authenticating-to-a-docker-registry

The plugin can only handle one repository. But I'm inconsistent. I don't think enterprise builds will go to hub.xenit.eu now.

File: tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/config/DefaultConfigurationProvider.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package eu.xenit.alfresco.tomcat.embedded.config;
```
Comment (todorinskiz) [inline tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/config/DefaultConfigurationProvider.java:1]: it's a little bit confusing to have some defaults in java, some in shell scripts. Can't we have it in the same place, like e.g. default alfresco-global.properties which is loaded first and then the env vars are used to override it ?

File: tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/valve/JsonAccessLogValve.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+public class JsonAccessLogValve extends ValveBase implements AccessLog {
+
+    private boolean requestAttributesEnabled = false;
+    private final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss,SSS");
+
+    @Override
+    public void log(Request request, Response response, long time) {
+        JsonObject root = new JsonObject();
+        root.addProperty("Time", simpleDateFormat.format(new Date()));
```
[the remark this answers is not in the document]
Comment (thijslemmens) [reply tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/valve/JsonAccessLogValve.java:22]: Let's compare this with access logging in Spring Boot. If we can have the same format, we can reuse dashboards.

File: tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/tomcat/TomcatFactory.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    private void addUserWithRole(Tomcat tomcat, String username, String password, String role) {
+        tomcat.addUser(username, password);
+        tomcat.addRole(username, role);
+    }
+
+    private void redirectLog4j(Path path) {
+        Path log4JPropertiesPath = path.resolve("WEB-INF/classes/log4j.properties");
+        Properties properties = new Properties();
+        try {
+            properties.load(Files.newBufferedReader(log4JPropertiesPath));
```
[the remark this answers is not in the document]
Comment (thijslemmens) [reply tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/tomcat/TomcatFactory.java:135]: What do you want to achieve?

File: .github/workflows/scheduled-build.yml [hunk 5]
```diff
@@ -8,29 +8,32 @@ jobs:
...
-       VERSIONS_TO_BUILD: ${{ matrix.version }}            
-       DOCKER_USER: ${{ secrets.DOCKER_USER }}
-       DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
-      run: ./gradlew -PrepoName=docker.io/xenit pushDockerImage
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Grant execute permission for gradlew
```
[the remark this answers is not in the document]
Comment (thijslemmens) [reply .github/workflows/scheduled-build.yml]: This was taken over from ci.yml. I don't think it is necessary, so I removed it.

File: tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/config/DefaultConfigurationProvider.java (hunk 2 above)
Comment (thijslemmens) [reply tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/config/DefaultConfigurationProvider.java:1]: which properties are in a shell script, other then the JAVA_OPTS ?

File: tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/valve/JsonAccessLogValve.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (thijslemmens) [reply tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/valve/JsonAccessLogValve.java:22]: I think we should open a broader discussion on the JSON logging format and how to align access logs, application logs and tomcat logs.

File: tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/config/DefaultConfigurationProvider.java (hunk 2 above)
Comment (thijslemmens) [reply tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/config/DefaultConfigurationProvider.java:1]: JAVA_OPTS need to be done before application start. I try to keep those props to a minimum.

File: tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/tomcat/TomcatFactory.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (thijslemmens) [reply tomcat-base/src/shared/java/eu/xenit/alfresco/tomcat/embedded/tomcat/TomcatFactory.java:135]: We do need to override the properties from the file

**Review by todorinskiz — APPROVED** [review]

**Pull request merged**